.ci/override-tekton-bundle.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+# This script will generate a new Tekton bundle that allows the overriding of images that are hard-coded into task steps
+TEMP_FOLDER=$WORKSPACE/tmp/bundle-override
+APPSTUDIO_QE_REPO=quay.io/redhat-appstudio-qe/test-images
+TASK_BUNDLE_IMG=$APPSTUDIO_QE_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA
+PIPELINE_BUNDLE_IMG=$APPSTUDIO_QE_REPO:pipeline-bundle-$JVM_BUILD_SERVICE_PR_SHA
+
+function getCurrentBuildBundle() {
+    curl -s -o "$TEMP_FOLDER"/build-kustomization.yaml https://raw.githubusercontent.com/redhat-appstudio/infra-deployments/main/components/build/kustomization.yaml
+
+    BUILD_BUNDLE=$(yq e ".configMapGenerator[].literals[]" \
+    "$TEMP_FOLDER"/build-kustomization.yaml \
+    | grep default_build_bundle \
+    | sed "s/default_build_bundle=//")
+
+    echo "New build bundle will be created based on $BUILD_BUNDLE"
+}
+
+function createPipelinesFile() {
+    pipelines=$(tkn bundle list "$BUILD_BUNDLE" | sed "s/pipeline.tekton.dev\///")
+    for pipeline in $pipelines; do
+        echo "---" >> "$TEMP_FOLDER"/pipelines.yaml
+        tkn bundle list -o yaml "$BUILD_BUNDLE" pipeline "$pipeline" >> "$TEMP_FOLDER"/pipelines.yaml
+    done
+}
+
+function createTaskFile() {
+    for bundle in $(grep bundle "$TEMP_FOLDER"/pipelines.yaml | sort -u | awk '{print $2}'); do 
+        tasks=$(tkn bundle list "$bundle" | sed "s/task.tekton.dev\///")
+
+        for task in $tasks; do
+            if [ "$task" == "s2i-java" ]; then
+                tkn bundle list -o yaml "$bundle" task "$task" >> "$TEMP_FOLDER"/task.yaml
+            fi
+        done
+    done
+}
+
+function updateAnalyzerImage() {
+    echo "jvm-build-service analyzer image set to $JVM_BUILD_SERVICE_ANALYZER_IMAGE"
+    yq e -i "select(.metadata.name == \"s2i-java\") \
+    | (.spec.steps[] | select(.name == \"analyse-dependencies-java-sbom\").image) \
+    |= \"$JVM_BUILD_SERVICE_ANALYZER_IMAGE\"" "$TEMP_FOLDER"/task.yaml
+}
+
+function updatePipelineRef() {
+    yq e -i "(.spec.tasks[].taskRef | select (.name == \"s2i-java\").bundle) \
+    |= \"$TASK_BUNDLE_IMG\"" "$TEMP_FOLDER"/pipelines.yaml
+}
+
+function createDockerConfig() {
+    mkdir ~/.docker
+    echo "$QUAY_TOKEN" | base64 -d > ~/.docker/config.json
+}
+
+function createAndPushTaskBundle() {
+    echo "Creating $TASK_BUNDLE_IMG"
+    tkn bundle push "$TASK_BUNDLE_IMG" -f "$TEMP_FOLDER"/task.yaml
+}
+
+function createAndPushPipelineBundle() {
+    echo "Creating $PIPELINE_BUNDLE_IMG"
+    tkn bundle push "$PIPELINE_BUNDLE_IMG" -f "$TEMP_FOLDER"/pipelines.yaml
+}
+
+mkdir -p "$TEMP_FOLDER"
```

Comment:
interesting, since you used `-p`, but this line if failing with 

`mkdir: cannot create directory '/go/src/github.com/redhat-appstudio/jvm-build-service/tmp/bundle-override': No such file or directory`

Maybe it is a permission issue masquerading as a no such file (i.e. the permission issue prevents you from creating `tmp` ??)

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/103/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1551861261765971968

Reply:
that `-p` was added recently

Reply:
The script worked in the latest run, the bundles were pushed as expected: https://quay.io/repository/redhat-appstudio-qe/test-images/manifest/sha256:41f0ded0364e0d234d99ba3613da39ceb9ac72a2b56195ce2f59083212724053

The error was some blip regarding Github, it returned a 429 message, not sure why:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/103/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-e2e/1551867931267174400

Reply:
hmm ok yeah looking at the time for https://github.com/redhat-appstudio/jvm-build-service/compare/45cb5b52ca7f7bb6e4708f681721ecace872e9c1..d95f519cae7121824084f105b509adaa2b546f88 vs. https://github.com/redhat-appstudio/jvm-build-service/pull/103#issuecomment-1195296295 it is feasible the test run was started before the force push

so let's see what happens with https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/103/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1551867931309117440

thanks